workbench-core/base/src/services/metadataService.test.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+import { MetadataService } from './metadataService';
+
+describe('metadata service', () => {
+  const metadataService = new MetadataService();
+  interface EntityA {
+    id: string;
+    sk: string;
+    pk: string;
+    productName: string;
+  }
+  interface EntityB {
+    id: string;
+    sk: string;
+    pk: string;
+    projectName: string;
+    projectType: string;
+  }
+  interface EntityAMetadata {
+    id: string;
+    sk: string;
+    pk: string;
+    projectName: string;
+    projectType: string;
+  }
+  interface EntityBMetadata {
```

Comment:
Do we need to declare another interface when these metadata interfaces match the shapes of the other interfaces?

Reply:
this is an example of metadata, they wont necessarily match the entity interfaces, I created separated ones to show how it would work, I will change the names to make it clear that metadata can contain data that is mapped from entities A and B